packages/lmi/src/lmi/llms.py

```diff
@@ -701,10 +736,60 @@ async def check_rate_limit(self, token_count: float, **kwargs) -> None:
                 **kwargs,
             )
 
+    async def _handle_refusal_via_fallback(
+        self, messages: list[Message], kwargs: dict[str, Any]
+    ) -> list[LLMResult]:
+        # Let's remove the current model from the configuration
+        current_model = self.name
```

Comment:
```suggestion
        with patch.object(self, "name", new_model_list[0].get("model_name") or ""):
            ...
```

Alternate way of avoiding `try`-`finally`. Feel free to ignore